crwatcher/status.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2017 the lostromos Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crwatcher
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"github.com/mitchellh/mapstructure"
+)
+
+type ResourcePhase string
+
+const (
+	PhaseNone     ResourcePhase = ""
+	PhaseApplying ResourcePhase = "Applying"
+	PhaseApplied  ResourcePhase = "Applied"
+	PhaseFailed   ResourcePhase = "Failed"
+)
+
+type ConditionReason string
+
+const (
+	ReasonUnknown               ConditionReason = "Unknown"
+	ReasonCustomResourceAdded   ConditionReason = "CustomResourceAdded"
```

Comment:
I think it might make sense to have a status for "noticed by the operator" instead of updated/added (the user is directly creating/updating the CR so that's not terribly useful status).

Maybe something like `ReasonInProgress`